packages/react/src/components/Forms/DatePicker/utils.ts

```diff
@@ -10,37 +13,75 @@ import {
 } from ":/components/Forms/DatePicker/types";
 import { DatePickerAuxSubProps } from ":/components/Forms/DatePicker/DatePickerAux";
 
-export const parseCalendarDate = (
+// These regexes are duplicated from
+// react-aria @internationalized/date
+const DATE_RE = /^(\d{4})-(\d{2})-(\d{2})$/;
+const DATE_TIME_RE =
+  /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?$/;
+const ABSOLUTE_RE =
+  /^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}))?(?::(\d{2}))?(?::(\d{2}))?(\.\d+)?(?:(?:([+-]\d{2})(?::?(\d{2}))?)|Z)$/;
+
+export const parseDateValue = (
   rawDate: StringOrDate | undefined
-): undefined | CalendarDate => {
+): undefined | DateValue => {
   if (!rawDate) {
     return undefined;
   }
   try {
+    // Parses an ISO 8601 date string, with no time components.
+    if (typeof rawDate === "string" && rawDate.match(DATE_RE)) {
+      return parseDate(rawDate);
+    }
+
+    // Parses an ISO 8601 date and time string, with no time zone.
+    if (typeof rawDate === "string" && rawDate.match(DATE_TIME_RE)) {
+      return parseDateTime(rawDate);
+    }
+
+    // Parses an ISO 8601 date and time string with a UTC offset.
+    if (typeof rawDate === "string" && rawDate.match(ABSOLUTE_RE)) {
+      return parseAbsoluteToLocal(rawDate);
+    }
+
+    // Parses the rawDate and converts it to the local time zone.
     const ISODateString = new Date(rawDate).toISOString();
-    return toCalendarDate(parseAbsoluteToLocal(ISODateString));
+    return parseAbsoluteToLocal(ISODateString);
   } catch (e) {
     throw new Error(
       "Invalid date format when initializing props on DatePicker component"
     );
   }
 };
 
-export const parseRangeCalendarDate = (
+export const parseRangeDateValue = (
   rawRange: StringsOrDateRange | undefined
 ): DateRange | undefined => {
   if (!rawRange || !rawRange[0] || !rawRange[1]) {
     return undefined;
   }
   return {
-    start: parseCalendarDate(rawRange[0])!,
-    end: parseCalendarDate(rawRange[1])!,
+    start: parseDateValue(rawRange[0])!,
+    end: parseDateValue(rawRange[1])!,
   };
 };
 
+export const isZonedDateTime = (object: any): object is ZonedDateTime => {
+  return !!object?.toAbsoluteString;
+};
+
+export const convertDateValueToString = (date: DateValue | null): string => {
+  // Unify the way we convert DateValue objets to string.
+  // If it's a ZonedDateTime instance, it returns an ISO 8601 UTC date string.
+  // else it returns an ISO 8601 date string with an optional time component.
+  return (
+    (isZonedDateTime(date) ? date?.toAbsoluteString?.() : date?.toString()) ||
```

Comment:
non-null assertion shouldn't be necessary on toAbsoluteString, `date?.toAbsoluteString()`.